From ce168ede29cedf06b5a35c9c5725a3f5555bbe32 Mon Sep 17 00:00:00 2001
From: David Gibson <dgibson@redhat.com>
Date: Mon, 15 Sep 2014 07:13:10 -0400
Subject: [virt] kvm/ppc: book3s/pr - Ignore PMU SPRs

Message-id: <1410765214-16377-55-git-send-email-dgibson@redhat.com>
Patchwork-id: 94741
O-Subject: [PATCH 54/78] KVM: PPC: Book3S PR: Ignore PMU SPRs
Bugzilla: 1123145 1123133 1123367
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Signed-off-by: David Gibson <dgibson@redhat.com>

commit f8f6eb0d189cf2724af5ebc8cad460c78fb1994e
Author: Alexander Graf <agraf@suse.de>
Date:   Tue Apr 22 12:41:06 2014 +0200

    KVM: PPC: Book3S PR: Ignore PMU SPRs

    When we expose a POWER8 CPU into the guest, it will start accessing PMU SPRs
    that we don't emulate. Just ignore accesses to them.

    Signed-off-by: Alexander Graf <agraf@suse.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kvm/book3s_emulate.c b/arch/powerpc/kvm/book3s_emulate.c
index 45d0a80..52448ef 100644
--- a/arch/powerpc/kvm/book3s_emulate.c
+++ b/arch/powerpc/kvm/book3s_emulate.c
@@ -455,6 +455,13 @@ int kvmppc_core_emulate_mtspr_pr(struct kvm_vcpu *vcpu, int sprn, ulong spr_val)
  case SPRN_WPAR_GEKKO:
  case SPRN_MSSSR0:
  case SPRN_DABR:
+#ifdef CONFIG_PPC_BOOK3S_64
+ case SPRN_MMCRS:
+ case SPRN_MMCRA:
+ case SPRN_MMCR0:
+ case SPRN_MMCR1:
+ case SPRN_MMCR2:
+#endif
   break;
 unprivileged:
  default:
@@ -553,6 +560,13 @@ int kvmppc_core_emulate_mfspr_pr(struct kvm_vcpu *vcpu, int sprn, ulong *spr_val
  case SPRN_WPAR_GEKKO:
  case SPRN_MSSSR0:
  case SPRN_DABR:
+#ifdef CONFIG_PPC_BOOK3S_64
+ case SPRN_MMCRS:
+ case SPRN_MMCRA:
+ case SPRN_MMCR0:
+ case SPRN_MMCR1:
+ case SPRN_MMCR2:
+#endif
   *spr_val = 0;
   break;
  default:
-- 
1.7.1